### PR TITLE
Revise PR template.

### PR DESCRIPTION
**Pull request by bdice**
## Description
I recently revamped our cuDF [CONTRIBUTING guide](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md). I would like to consider replacing the current PR template (which has a fairly daunting amount of text that is immediately deleted by many contributors) with a short checklist of actionable items and a reference to the CONTRIBUTING guide for longer content.

I kept this draft very minimal. Reviewers can see other examples here for inspiration: https://axolo.co/blog/p/part-3-github-pull-request-template. Happy to crowdsource others' thoughts here.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (wence-) [inline .github/PULL_REQUEST_TEMPLATE.md:48]: This part of the advice arguably could be ported into `CONTRIBUTING.md`. This seems like useful workflow information for contributors (especially new ones). IIUC, cudf uses squash-merges for PRs, so fixing up commits in response to review is not necessary for a clean commit history (and, as noted here, can hinder in the case of multiple rounds of reivew).

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (jrhemstad) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: Two additional things came to mind. Feel free to push back on these ideas.

```suggestion
## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] New or existing benchmarks cover these changes.
- [ ] The documentation is up to date with these changes.
- This change is:
   - [ ] Breaking
   - [ ] Non-breaking
```

**Review by wence- — COMMENTED** [review]
Could possibly port some of the culled text into contributing?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (jrhemstad) [inline .github/PULL_REQUEST_TEMPLATE.md:2]:
```suggestion
<!-- Standalone description of changes (will be included in CHANGELOG). Reference any issues closed by this PR with "closes #1234" or "fixes #5678" -->
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (bdice) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Benchmarks I can support -- with the caveat that C++ has the best benchmarking at present, and the Python is somewhat outdated.

Breaking/non-breaking is already handled by labels, and the CI requires us to label one or the other. I don't think it is necessary to input that information twice.

Comment (shwina) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: IMO tests are going to be much more frequent additions than benchmarks (the latter is typically for new features)

Comment (bdice) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Ooh, interesting. I hadn't considered that. I'm used to the GitHub default message for squash merges, which looks like

```
Title of PR. (https://github.com/rapidsai/cudf/pull/10774)

* First commit message

* Second commit message

* Third commit message
```

Without calling out any specific commits, the existing (long!) PR template comment can be found several times in the output of `$ git log | grep "Thank you for contributing"`. At least this is shorter?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (wence-) [reply .github/PULL_REQUEST_TEMPLATE.md:48]: I think that if it's outdated then there's no real reason to include it. I agree that github seems to be better than a few years ago at handling this kind of code movement (I don't recall having comments resolved by rebase and github losing track for a while).

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (bdice) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: I would strongly discourage duplicating information in the template and labels. The overhead of every checkbox / button adds up. However, I do agree that our categories are hard to locate. This is the official reference, but I don't think it's linked anywhere in the cuDF repository: https://docs.rapids.ai/resources/label-checker/#summary

It would be helpful if the output message said "Missing category label (bug, doc, feature request, improvement)".

Though on second thought, why is "feature request" a supported pull request category? Sounds like it should only apply to issues...

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (bdice) [reply .github/PULL_REQUEST_TEMPLATE.md:48]: As I think about this more, this suggests an opportunity to rework some of the CONTRIBUTING guide. The differentiation between "[Your First Issue](https://github.com/rapidsai/cudf/blob/ad126065109aaa72b6eb324ba5abd555b70bb4ae/CONTRIBUTING.md#your-first-issue)" and "[Seasoned Developers](https://github.com/rapidsai/cudf/blob/ad126065109aaa72b6eb324ba5abd555b70bb4ae/CONTRIBUTING.md#seasoned-developers)" is irrelevant and a little presumptive. It'd be clearer to write something like "Development workflow" and "Project planning and organization", and incorporate tips about draft PRs, etc. To keep this focused, I will create a separate PR for the CONTRIBUTING guide before resolving this thread.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (shwina) [reply .github/PULL_REQUEST_TEMPLATE.md]: Let's be consistently imperative here:

```
<!-- Provide a standalone description of changes in this PR (will be included in CHANGELOG). -->
<!-- Reference any issues closed by this PR with "closes #1234". -->
```

Is the description for each PR actually included in CHANGELOG? I don't see them: https://github.com/rapidsai/cudf/blob/branch-22.06/CHANGELOG.md

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (karthikeyann) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: Adding to this, another idea is to add "category" to description text of the label. so that if I type "category", it will list only labels with "category" text in their description, and I can choose them easily.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (bdice) [reply .github/PULL_REQUEST_TEMPLATE.md]: You're correct. The _title_ is included in the changelog. I do think that's helpful to note.

Comment (bdice) [reply .github/PULL_REQUEST_TEMPLATE.md]: Updated in 2cf27046840523ed61471bb981fe23c8d96551e5. Feel free to open a new thread for further suggestions. This one is outdated so I'll resolve it.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (bdice) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: I like this idea. `category: doc` is longer, but it's a lot easier to distinguish the special category labels.

**Comment by bdice** [comment]
@jrhemstad @shwina @wence- @karthikeyann Thank you very much for your thoughtful input! I opened this PR for review. I have two action items in mind, outside the scope of this PR:
- Update the CONTRIBUTING guide with some of the lost information from this PR template
- Work with @karthikeyann and the ops team to adapt the bot's squash commit messages to filter out the checklist

For now, I'm seeking approval on the language of the PR template. I will hold off on merging this until we have the aforementioned items complete (or in review).

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (shwina) [inline .github/PULL_REQUEST_TEMPLATE.md:2]: Sorry for the nit: "standalone", as opposed to what?

Comment (bdice) [reply .github/PULL_REQUEST_TEMPLATE.md:2]: @jrhemstad suggested this word (feel free to voice your thoughts, Jake). I agree it may not be needed.
```suggestion
<!-- Provide a description of changes in this PR. -->
```

Comment (davidwendt) [reply .github/PULL_REQUEST_TEMPLATE.md:2]: > Sorry for the nit: "standalone", as opposed to what?

I think as opposed to just linking to an issue for example.

**Review by wence- — APPROVED** [review]
I think the proposed changes look good. If wanting to include information about benchmark/test coverage perhaps the contributing guide should say how to check? I think codecov will comment on code that is not covered by the tests, but I don't know how to check that benchmarks cover what I add (and I did look!)

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (jrhemstad) [reply .github/PULL_REQUEST_TEMPLATE.md:2]: > I think as opposed to just linking to an issue for example.

Exactly. Far too many PRs still do this.

"Self-contained" could also be used in lieu of "standalone" if that's preferred. 

The main idea is that it shouldn't rely on referencing anything else to understand the gist of what the PR does.

**Review by jrhemstad — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (karthikeyann) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: I am not suggesting to include "category" in label itself (also a good idea). Just in description text of the label. That is enough for search to filter "category" when we type "category".

**Review by karthikeyann — APPROVED** [review]
LGTM 👍

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (bdice) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: Aha! The description would be a good compromise because it doesn’t require changing bot logic.

Comment (davidwendt) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: Have you started CI issue to filter the checklist from the description? I would think that work would perhaps need to be done before merging this PR.

Comment (karthikeyann) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: could @rapidsai/admins or @rapidsai/cudf-admin update this? 
add "category" to the description text of the following labels: bug, doc, feature request, improvement.
This allows easily identifying the category labels during PR creation, by simply typing "category" in label box.

Comment (karthikeyann) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: https://github.com/rapidsai/ops-bot/issues/102
That's not really required before this PR merge. Even if the commit text contains this checklist, it's not a big issue. Although we could request the developers to remove the checklist before doing gpucibot merge.

@bdice does it make sense to add another point to delete the checklist before merging?

**Comment by karthikeyann** [comment]
> I don't know how to check that benchmarks cover what I add (and I did look!)

benchmarks are not run as part of CI right now.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (harrism) [reply .github/PULL_REQUEST_TEMPLATE.md:48]: https://stackoverflow.com/questions/50132325/github-advises-force-pushing-can-corrupt-your-pull-request-why

Comments are not lost when you force push, but they can be disassociated with the code when the commit they refer to is lost. You can no longer see the code they are referring to when you click on the file link at the top of the comment.  

I believe it would also be possible to lose changes from other contributors or changes made via the github UI (e.g. via code review suggestions). 

This is why I recommended we have a rule against force pushing once a PR is under review.  I believe that rule/suggestion should continue to be documented.

As the above linked SO answer suggests, 
> When you really need to do a force push, you can always create a new pull request, which is much safer and keeps away annoying problems

Comment (harrism) [reply .github/PULL_REQUEST_TEMPLATE.md:48]: I wish these templates and docs were RAPIDS-wide rather than specific to cuDF.

**Review by jrhemstad — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (bdice) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: I would vote to take no action here, and accept that the checklist will be included in squash commit messages. It’s not really a problem in my eyes.

**Comment by ajschmidt8** [comment]
FYI, I just merged @karthikeyann's PR below.

- https://github.com/rapidsai/ops-bot/pull/103

**Review by wence- — APPROVED** [review]

**Comment by karthikeyann** [comment]
rerun tests

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (bdice) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: Update: https://github.com/rapidsai/ops-bot/pull/103/ was merged, so only the contents of `## Description` will be used for the PR summary. That's a nice feature, thanks @karthikeyann / @ajschmidt8!

**Review by bdice — COMMENTED** [review]
I'm going to merge this, since it has 4 approvals. I will leave the adjacent conversations open, which are not in scope for this PR.

**Comment by bdice** [comment]
@gpucibot merge

**Pull request merged**
